dashboard/test/ui/features/support/browser_helpers.rb

```diff
@@ -20,6 +22,33 @@ def element_has_i18n_text(selector, language, loc_key)
     text.should eq response
   end
 
+  # This function checks that the text within the given selector matches the
+  # text produced by rendering the markdown given by the localization string.
+  #
+  # This is used to test that localized markdown is being rendered in any given
+  # container element.
+  def element_has_i18n_markdown(selector, language, loc_key)
+    loc_key.gsub!('\"', '"')
+
+    # Grab html of the markdown container from the browser.
+    html = @browser.execute_script("return $(\"#{selector}\").html();")
+
+    # Get localized markdown from server.
+    response = HTTParty.get(replace_hostname("http://studio.code.org/api/test/get_i18n_t?key=#{loc_key}&locale=#{language}")).parsed_response
```

Comment:
Purely out of curiosity: How did you find this endpoint? I never knew we had it!

Reply:
It is used in the normal `element_has_i18n_text` function above it. The `TestController` in `dashboard/app/controllers/test_controller.rb` likely only exists in the test environment. Very useful though!